src/backend/distributed/utils/citus_stat_tenants.c

```diff
@@ -121,7 +122,10 @@ citus_stat_tenants_local(PG_FUNCTION_ARGS)
 
 	for (int tenantIndex = 0; tenantIndex < monitor->tenantCount; tenantIndex++)
 	{
-		UpdatePeriodsIfNecessary(&monitor->tenants[tenantIndex], monitoringTime);
+		/* this is not a tenant query, it's a query to get stats */
+		bool isTenantQuery = false;
```

Comment:
minor: I'd maybe be a bit more literal and say something like `bool setLastQueryTime`, or set lastQueryTime outside of UpdatePeriodsIfNecessary.

Conceptually, updating lastQueryTime seems more like a job for RecordTenantStats.

Reply:
Makes sense to put it in `RecordTenantsStats`, so I can remove the check.